### PR TITLE
Add username modal component

### PR DESCRIPTION
**Pull request by tobbe-j**
Adds a component for requesting and saving a username for users.

Also now displays usernames in comments for users with username.

### CODE REVIEW

**Comment by tobbe-j** [comment]
[(0/5) [story] [comment view] As a feedback giving user, after writing and sending a comment, the Feedbacker UI asks for a name if no name is specified, so that I can write my own name or nickname and use name Tobias](https://trello.com/c/tkkTMjEe/104-0-5-story-comment-view-as-a-feedback-giving-user-after-writing-and-sending-a-comment-the-feedbacker-ui-asks-for-a-name-if-no-nam)

**Comment by tobbe-j** [comment]
Might still need some configuration to be usable in different places, currently optimized for using when adding new comment

File: client/src/components/add-username-modal/add-username-modal.scss [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+.username-modal {
+  @include z-level(3);
+  display: flex;
+  flex-flow: column;
+  justify-content: space-around;
+  position: fixed;
+  right: $baseline;
+  bottom: $baseline;
+  width: 35vw;
```
Comment (HenrikHelenius) [inline client/src/components/add-username-modal/add-username-modal.scss]: @tobbe-j has this been tested at mobile? consider using max-width, min-width and same with heights

File: client/src/components/add-username-modal/add-username-modal.scss [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  border-radius: $border-radius-default;
+  background-color: $palette-lighter;
+  cursor: default;
+  &:focus {
+    outline: none;
+    box-shadow: 0 2px 10px $palette-shadow-default;
+  }
+}
+
+.add-button {
```
Comment (HenrikHelenius) [inline client/src/components/add-username-modal/add-username-modal.scss]: cahnge to input[type="submit"], could it be extended from some place => even DRY:er?

File: client/src/components/add-username-modal/add-username-modal.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  padding: 1em 2em;
+  input[type="text"] {
+    @extend .comment;
+    display: block;
+    min-width: 100%;
+    border: none;
+  }
+}
+
+.username-overlay {
```
Comment (HenrikHelenius) [inline client/src/components/add-username-modal/add-username-modal.scss:54]: not sure if this should be lifted to higher abstraction later, as @sofiatengvall uses similar stuff.. @tobbe-j

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

**Review by quvide — COMMENTED** [review]
Backend changes look ok

File: client/src/components/add-username-modal/add-username-modal.scss (hunk 3 above)
Comment (tobbe-j) [reply client/src/components/add-username-modal/add-username-modal.scss:54]: Could probably be left as is in this commit, and then lifted/moved in feature-intro branch where the same is used

File: client/src/components/add-username-modal/add-username-modal.scss (hunk 2 above)
Comment (tobbe-j) [reply client/src/components/add-username-modal/add-username-modal.scss]: Couldn't find similar uses in this version, can this be done in later branches where more textboxes are i use?

File: client/src/components/add-username-modal/add-username-modal.scss (hunk 1 above)
Comment (tobbe-j) [reply client/src/components/add-username-modal/add-username-modal.scss]: Should now work decently on narrower screens too

File: client/src/components/add-username-modal/add-username-modal.scss (hunk 3 above)
Comment (HenrikHelenius) [reply client/src/components/add-username-modal/add-username-modal.scss:54]: yes

**Review by HenrikHelenius — DISMISSED** [review]
nothing to add

File: client/src/components/add-username-modal/add-username-modal.scss (hunk 2 above)
Comment (HenrikHelenius) [reply client/src/components/add-username-modal/add-username-modal.scss]: fixable later

**Review by HenrikHelenius — APPROVED** [review]
🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐🦐

**Pull request merged**